user_guide_src/source/installation/upgrade_4xx.rst

```diff
@@ -1,114 +1,102 @@
-
-Nothing should be really broken by this change, but if your application
-relies on the default value, you should double-check it and either
-explicitly set the desired format, or adapt your front-end to use proper
-HTML 5 formatting.
\ No newline at end of file
+CodeIgniter 4 is a rewrite of the framework, and is not backwards compatible.
+It is more appropriate to think of converting your app, rather than upgrading it.
+Once you have done that, upgrading from one version of CodeIgniter 4 to the next
+will be straightforward.
+
+The "lean, mean and simple" philosophy has been retained, but the 
+implementation has a lot of differences, compared to CodeIgniter 3.
+
+There is no 12-step checklist for upgrading. Instead, start with a copy
+of CodeIgniter 4 in a new project folder, `however you wish to install and 
+use it </installation>`_, 
+and then convert and integrate your app components.
+We'll try to point out the most important considerations here.
+
+Not all of the CI3 libraries have been ported or rewritten for CI4!
+See the threads in the `CodeIgniter 4 Roadmap <https://forum.codeigniter.com/forum-33.html>`_
+subforum for an uptodate list!
+
+**Downloads**
+- CI4 is still available as a ready-to-run zip or tarball, which
+includes the user guide (though in the `docs` subfolder
+- It can also be installed using Composer
+
+**Namespaces**
+- CI4 is built for PHP7.1+, and everything in the framework is namespaced, even the helpers
+
+**Application Structure**
+- The framework still has `application` and `system` folders, with the same 
+interpretation as before
+- The framework now provides for a `public` folder, intended as the document
+root for your app
+- There is also a `writable` folder, to hold cache data, logs, and session data
+- The `application` folder looks very similar to that for CI3, with some
+name changes (eg Filters instead of hooks) and capitalization, and some subfolders
+moved to the `writable` folder
+- There is no longer a nested `application/core` folder, as we have
+a different mechanism for extending framework components (see below)
+
+**Class loading**
+- There is no longer a CodeIgniter superobject, with framework component
+references magically as properties of your controller
+- Classes are instantiated where needed, and components are managed
+by `Services`
+- The class loader automatically handles PSR4 style class locating,
+within the `App` (application) and `CodeIgniter` (i.e. system) top level
+namespaces; with composer autoloading support, and even using educated
+guessing to find your models and libraries if they are in the right
+folder even though not namespaced
+- You can configure the class loading to support whatever application structure
+you are most comfortable with, including the "HMVC" style
+
+**Controllers**
+- Controllers extend \CodeIgniter\Controller instead of CI_Controller
+- They don't use a constructor any more (to invoke CI "magic"); they
```

Comment:
I wouldn't mention `initController`. That is used behind the scenes to free up the constructor, but the user's don't need to know about it. The ones that know about it so far are just confused by it :)

Reply:
Will adjust wording